app/jsx/pages/BrowsePage.jsx

```diff
@@ -60,6 +64,90 @@ felis tellus mollis orci, sed start a journal nunc eget odio.
       <br/><br/>
     </div>
   )}
+
+}
+
+// Tree traversal help via http://codepen.io/anon/pen/Ftkln
+// 'depts' property (JSON)
+// [ ...,
+//  {"id"=>"international",
+//   "name"=>"UCLA International Institute",
+//   "children"=>
+//    [{"id"=>"asia",
+//      "name"=>"UCLA Asia Institute",
+//      "children"=>[{"id"=>"ccs", "name"=>"UCLA Center for Chinese Studies"}]}, .. ], } .. ]
+class DeptTree extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { depts: this.props.depts }
+    // this.onSelect = this.onSelect.bind(this)
+  }
+  onSelect(node) {
+    if (this.state.selected && this.state.selected.isMounted()) {
+      this.state.selected.setState({selected: false})
+    }
+    this.setState({selected: node})
+    node.setState({selected: true})
+    if (this.props.onCategorySelect) {
+      this.props.onCategorySelect(node)
+    }
+  }
+  render() { return (
+    <div>
+      <h2>Departments</h2>
+      <p>
+Lorem ipsum dolor sit amet
+
+consectetur adipiscing elit. Aenean euismod bibendum laoreet. Proin gravida dolor sit amet lacus accumsan et viverra justo commodo. Proin sodales pulvinar tempor. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Nam fermentum, nulla luctus pharetra vulputate,
+
+felis tellus mollis orci, sed start a journal nunc eget odio.
+      </p>
+      Expand All<br/>
+      <ul>
+        { this.props.depts.map((node) =>
+            <TreeNode key={node.id} depts={node} onCategorySelect={this.onSelect}/>) }
+      </ul>
+    </div>
+  )}
+}
+
+class TreeNode extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = { children: this.props.depts.children }
+    this.onCategorySelect = this.onCategorySelect.bind(this)
+    this.onChildDisplayToggle = this.onChildDisplayToggle.bind(this)
+  }
+  onCategorySelect(ev) {
+    if (this.props.onCategorySelect) {
+      this.props.onCategorySelect(this)
+    }
+    ev.preventDefault()
+    ev.stopPropagation()
+  }
+  onChildDisplayToggle(ev) {
+    if (this.props.depts.children) {
+      if (this.state.children && this.state.children.length) {
+        this.setState({children: null})
+      } else {
+        this.setState({children: this.props.depts.children})
+      }
+    }
+    ev.preventDefault()
+    ev.stopPropagation()
+  }
+  render() {
+    if (!this.state.children) this.state.children = []
+    return (
```

Comment:
When I run this code, no matter what I click on (label or the little bullet) both do a child display toggle. The <Link> never activates for me. Is that on your roadmap?